### PR TITLE
KOTOR: Border color

### PR DESCRIPTION
**Pull request by Nostritius**
This PR implements border colors for kotor and kotor2 widgets

### CODE REVIEW

**Comment by DrMcCoy** [comment]
Some before/after pictures would have been nice.

From what I can see, in KotOR2:

20180515t121756
20180515t122005

20180515t121805
20180515t122013

And KotOR:

20180515t121731
20180515t122144

Not sure the tab coloring on the unselected tabs is correct, though?

There's also this, which happens if you close the ingame menu and then reopen the same tab you closed on:

20180515t122245

**Comment by DrMcCoy** [comment]
(The PR looks okay otherwise to me, though)

**Comment by DrMcCoy** [comment]
Oh, right, "kolors" in the commit message, though, should be "colors" :P

**Comment by DrMcCoy** [comment]
Also, could you start capitalising class names in the commit message correctly as well? I.e. "BorderQuad" instead of "borderquad". So that it's clearer what you're talking about

**Comment by Nostritius** [comment]
All issues should now be fixed. This includes the tab problem, i added handling if the border defines no color.

File: src/engines/kotor/gui/widgets/kotorwidget.cpp [hunk 1]
```diff
@@ -328,6 +330,10 @@ void KotORWidget::load(const Aurora::GFF3Struct &gff) {
 
 	if (border.fill.empty())
 		_quad->setColor(0.0f, 0.0f, 0.0f, 0.0f);
+	else if (border.r == 0 && border.g == 0 && border.b == 0)
+		_quad->setColor(1.0f, 1.0f, 1.0f, 1.0f);
```
Comment (DrMcCoy) [inline src/engines/kotor/gui/widgets/kotorwidget.cpp]: That's a bit...weird. So if the file specifies black, it's actually not?

Comment (Nostritius) [reply src/engines/kotor/gui/widgets/kotorwidget.cpp]: Would it be better if i add a border.hasColor Flag?

Comment (DrMcCoy) [reply src/engines/kotor/gui/widgets/kotorwidget.cpp]: Hmm, we need a better way to distinguish no between "no color defined" and "color defined, but it's black" here.

`KotORWidget::createBorder()` (which is the method filling in the border struct, yes?) should be extended. `struct Border` should get a `bool hasColor`, set to the return value of `b.hasField("COLOR")`.

**Comment by Nostritius** [comment]
I have now changed the code to use a hasColor Flag.

**Comment by DrMcCoy** [comment]
Merged as c19522df...b99404cc, thanks! :)

**Pull request closed without merging**